### DISCUSSION TITLE
Automatically gather .js or .scss files with webpack?

### DISCUSSION THREAD

**Discussion by Anoerak** [symfony/symfony#50410] [Q&A] [answered]
Hello,

I started to use **php** and **Symfony** back in January and still have a lot to figure out.
Right now, I want to ask something about Encore and the best way to manage _.scss_ and _.js_ files.

I organize my folder like this :
```
templates
| components
| | header
| | | header.html.twig
| | | header.js
| | | header.scss
```

I configured **webpack** and got everything to work just fine but I was wondering (and maybe i did something wrong here) if it was possible to automatically gather all _.js_ files and all _.scss_ files from the different templates.

As it is now, I have to manually require or import my files in my “_assets/app.js_” and “_assets/app.scss_”.
```scss
@import '~/assets/styles/abstracts/mixins.scss';
@import '~/assets/styles/abstracts/variables.scss';
@import '~/assets/styles/base/base.scss';
@import '~/assets/styles/base/fonts.scss';
@import '~/templates/components/header/header.scss';
```
Am I doing something I shouldn’t do or is it the way it is supposed to work?

Thanks for your help.

**Comment by stof** [comment, marked as answer]
Sass does not have a feature to import by globs.

Regarding webpack, maybe there are loaders to do that. But you have more chances to get an answer by asking for support in the webpack community.

**Reply by Anoerak** [reply]
Thanks,
I'm gonna look into that :)

**Comment by WebMamba** [comment]
Hey, @Anoerak what you're doing it's totally legit! Even though, most people using sass use special Architecture to organize their imports. There are a lot of famous sass architecture formats, but the most famous one is the 7-1 https://sass-guidelin.es/#architecture. The idea is to have 7 folders, to put all your scss files and one (usually main.scss) in charge of importing everything.

**Reply by Anoerak** [reply]
Thanks,
I'm gonna look into @stof suggestion about Webpack. The idea would be to gather all my _scss_ files and _js_ files from my **templates** into my main file _app.js_ and _app.scss_ in my **asset** folder. 
A bit like what is done with the unit test, you launch your command and it gathers all the test files based on the extension.
